packages/authentication/index.js

```diff
@@ -1,7 +0,0 @@
-// Copyright IBM Corp. 2017,2018. All Rights Reserved.
-// Node module: @loopback/authentication
-// This file is licensed under the MIT License.
-// License text available at https://opensource.org/licenses/MIT
-
-const nodeMajorVersion = +process.versions.node.split('.')[0];
-module.exports = nodeMajorVersion >= 7 ? require('./dist') : require('./dist6');
```

Comment:
With Node.js 10.0.0 coming in few months, I think we should start planning how to transpile to es2018 target. Since transpiling for both Node.js 8.x and 10.x will require these index files around, I think we should preserve them.

https://twitter.com/jasnell/status/958765477089632257
> PSA: Mark your calendars, the expected release date for Node.js 10.0.0 is on or around April 24th. Expect the first As-Close-To-A-Release-Candidate-As-We-Can-Get-It to hit in early April.

Thoughts?

Reply:
(I do realize this contradicts the checklist in https://github.com/strongloop/loopback-next/pull/945, which I created back in October last year.)

Reply:
After thinking about this a bit more: we should leave out transpilation target `es2018` out of scope of this pull request.

If we need to keep `index.js` file around because of https://github.com/strongloop/loopback-next/pull/945#discussion_r165288862, then I am ok with simplifying it to a simple redirect to `dist`. If it turns out TS+VSCode works well without this hand-written `index.js` around, then I am ok with deleting it as is proposed here.

Reply:
Actually yeah we were discussing about es2018 before making this change. I think it might be better to keep them as they do no harm, and if used in the future, we don't have to remake them.